cluster/src/main/java/org/apache/iotdb/cluster/log/manage/serializable/SyncLogDequeSerializer.java

```diff
@@ -945,23 +1008,34 @@ private int comparePersistLogFileName(File file1, File file2) {
     if (endIndex - startIndex > maxNumberOfLogsPerFetchOnDisk) {
       newEndIndex = startIndex + maxNumberOfLogsPerFetchOnDisk;
     }
-    logger.debug("intend to get logs between[{}, {}], actually get logs between[{},{}]", startIndex,
-        endIndex, startIndex, newEndIndex);
+    logger
+        .debug("intend to get logs between[{}, {}], actually get logs between[{},{}]", startIndex,
+            endIndex, startIndex, newEndIndex);
 
-    List<Pair<File, Pair<Long, Long>>> logDataFileAndOffsetList = getLogDataFileAndOffset(
-        startIndex, newEndIndex);
-    if (logDataFileAndOffsetList.isEmpty()) {
-      return Collections.emptyList();
-    }
+    // maybe the logs will be delete during checkDeletePersistRaftLog or clearAllLogs,
+    // use writeLock for two reasons:
+    // 1.if the log file to read is the last log file, we need to get write lock to flush logBuffer,
+    // 2.avoid log files be deleted
+    lock.writeLock().lock();
```

Comment:
Is there anywhere using read lock of this lock? If not, maybe it is better to consider using a more light-weighted lock.

Reply:
i intend to use ReentrantLock instead of ReentrantReadWriteLock